### PR TITLE
Sniff service protocol port from device

### LINKED ISSUE #1988: Atom ⌘-R fails if there's another Flutter app running in background on the Android device
It complains that the Observatory port is in use and shows a black screen (because Atom launches the app with --start-paused)

### PR DESCRIPTION
**Pull request by johnmccutchan**
Fixes #1988

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@abarth @devoncarew

**Comment by johnmccutchan** [comment]
@abarth mostly for how you want the other part of this change to land in sky_engine.

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -20,6 +20,7 @@ import '../globals.dart';
 import '../toolchain.dart';
 import 'adb.dart';
 import 'android.dart';
+import '../service_protocol.dart';
```
Comment (abarth) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Please alphabetize

**Comment by devoncarew** [comment]
lgtm

(in atom you can sort directives using cmd-option-o)

**Comment by johnmccutchan** [comment]
Related https://github.com/flutter/engine/pull/2515

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Done.

**Comment by johnmccutchan** [comment]
@chinmaygarde for iOS changes.

**Pull request merged**